script/_systemd_networkd

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+#
+#  Copyright (c) 2025, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+ETH_INTERFACE="eth0"
+ETH_NETWORK_FILE="/etc/systemd/network/10-${ETH_INTERFACE}.network"
+
+WIFI_INTERFACE="wlan0"
+WIFI_NETWORK_FILE="/etc/systemd/network/25-${WIFI_INTERFACE}.network"
+
+create_network_file()
+{
+    local interface="$1"
+    local network_file="$2"
+
+    # Create the network configuration file for the network interface
+    {
+        cat <<EOF
+[Match]
+Name=${interface}
+
+[Network]
+DHCP=yes
+IPv6AcceptRA=yes
+EOF
+        # If OpenThread's built-in DHCPv6 PD client is enabled, we must
+        # disable systemd-networkd's client to avoid port conflicts.
+        if with OT_BORDER_ROUTING_DHCP6_PD_CLIENT; then
```

Comment:
I doubt if this will work as what you expect. The `OT_BORDER_ROUTING_DHCP6_PD_CLIENT` is a CMake build option while you're checking it as a variable in bash.